[Remote clusters] Migrate server code out of legacy

alisonelizabeth opened the pull request:
This PR completes the NP migration for the `remote_clusters` **server-side** code. I will address the client code in a separate PR.

Changes include:
- Adopted NP router
   - I removed a [workaround in update_route.ts](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/remote_clusters/server/routes/api/update_route.ts#L33) as the issue has been fixed on the ES side.
- Adopted NP license plugin
  - I'm happy to change the implementation here once we decide on a common pattern.
- Updated route tests using the NP core mocks
- Moved server code out of `legacy` directory

_Note: It may be helpful to review the code changes by commit._

### How to test
You will need to set up two clusters in order to test remote clusters. For example:

    yarn es snapshot
    yarn es snapshot -E cluster.name="my_cluster" -E http.port=9201 -E path.data=/tmp/es (in another tab)

- Test all routes and confirm behaving as expected (create, edit, delete, get remote clusters)
- Test license check (remote clusters is available with `Basic` license)

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/)
* Commit: 03d8aa15abdbe5914f8b5604cdf27269b95977ef

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/testReport)

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/legacy/plugins/remote_clusters/server/routes/api.[API Routes] Remote Clusters deleteHandler() encountered a declaration exception</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/testReport//junit/X-Pack%20Jest%20Tests/x-pack_legacy_plugins_remote_clusters_server_routes_api/Kibana_Pipeline___x_pack_intake_agent____API_Routes__Remote_Clusters_deleteHandler___encountered_a_declaration_exception)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches


```

#### Stack Trace
```
TypeError: (0 , _delete_route.createDeleteHandler) is not a function
    at Suite.Object.<anonymous>.describe (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/remote_clusters/server/routes/api/delete_route.test.ts:16:25)
    at addSpecsToSuite (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/Env.js:496:51)
    at Env.describe (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/Env.js:466:11)
    at describe (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/jasmineLight.js:81:18)
    at Object.<anonymous> (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/remote_clusters/server/routes/api/delete_route.test.ts:12:1)
    at Runtime._execModule (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runtime/build/index.js:867:68)
    at Runtime._loadModule (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runtime/build/index.js:577:12)
    at Runtime.requireModule (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runtime/build/index.js:433:10)
    at /var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:202:13
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:27:24)
    at _next (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:47:9)
    at /var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:52:7
    at new Promise (<anonymous>)
    at /var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:44:12
    at jasmine2 (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:60:19)
    at /var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runner/build/runTest.js:385:24
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runner/build/runTest.js:161:24)
    at _next (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runner/build/runTest.js:181:9)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

---

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/legacy/plugins/remote_clusters/server/routes/api.[API Routes] Remote Clusters updateHandler() returns the cluster information from Elasticsearch</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/testReport//junit/X-Pack%20Jest%20Tests/x-pack_legacy_plugins_remote_clusters_server_routes_api/Kibana_Pipeline___x_pack_intake_agent____API_Routes__Remote_Clusters_updateHandler___returns_the_cluster_information_from_Elasticsearch)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches


```

#### Stack Trace
```
TypeError: (0 , _update_route.updateHandler) is not a function
    at Object.it (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/remote_clusters/server/routes/api/update_route.test.ts:41:28)
    at Object.asyncJestTest (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at resolve (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:43:12)
    at new Promise (<anonymous>)
    at mapper (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at promise.then (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:73:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

---

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/legacy/plugins/remote_clusters/server/routes/api.[API Routes] Remote Clusters updateHandler() throws an error if the response doesn't contain cluster information</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/testReport//junit/X-Pack%20Jest%20Tests/x-pack_legacy_plugins_remote_clusters_server_routes_api/Kibana_Pipeline___x_pack_intake_agent____API_Routes__Remote_Clusters_updateHandler___throws_an_error_if_the_response_doesn_t_contain_cluster_information)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches


```

#### Stack Trace
```
TypeError: (0 , _update_route.updateHandler) is not a function
    at Object.it (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/legacy/plugins/remote_clusters/server/routes/api/update_route.test.ts:73:7)
    at Object.asyncJestTest (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at resolve (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:43:12)
    at new Promise (<anonymous>)
    at mapper (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at promise.then (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:73:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

---
and 10 more failures, only showing the first 3.

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/","number":"24390","commit":"03d8aa15abdbe5914f8b5604cdf27269b95977ef"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/)
* Commit: 0021d90befbbd1f2ebdaabb6040435d4c9329cba

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/testReport)

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/plugins/remote_clusters/server/routes/api.[API Routes] Remote Clusters getAllHandler() converts the ES response object to an array</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/testReport//junit/X-Pack%20Jest%20Tests/x-pack_plugins_remote_clusters_server_routes_api/Kibana_Pipeline___x_pack_intake_agent____API_Routes__Remote_Clusters_getAllHandler___converts_the_ES_response_object_to_an_array)


#### Stack Trace
```
TypeError: (0 , _get_route.getAllHandler) is not a function
    at Object.it (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/plugins/remote_clusters/server/routes/api/get_route.test.ts:22:28)
    at Object.asyncJestTest (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at resolve (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:43:12)
    at new Promise (<anonymous>)
    at mapper (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at promise.then (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:73:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

---

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/plugins/remote_clusters/server/routes/api.[API Routes] Remote Clusters getAllHandler() returns an empty array when ES responds with an empty object</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/testReport//junit/X-Pack%20Jest%20Tests/x-pack_plugins_remote_clusters_server_routes_api/Kibana_Pipeline___x_pack_intake_agent____API_Routes__Remote_Clusters_getAllHandler___returns_an_empty_array_when_ES_responds_with_an_empty_object)


#### Stack Trace
```
TypeError: (0 , _get_route.getAllHandler) is not a function
    at Object.it (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/plugins/remote_clusters/server/routes/api/get_route.test.ts:36:28)
    at Object.asyncJestTest (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at resolve (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:43:12)
    at new Promise (<anonymous>)
    at mapper (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at promise.then (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/queueRunner.js:73:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

---

<details><summary>Kibana Pipeline / kibana-xpack-agent / X-Pack API Integration Tests.x-pack/test/api_integration/apis/management/remote_clusters/remote_clusters·js.apis management remote clusters Remote Clusters Add should not allow us to re-add an existing remote cluster</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/testReport//junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_management_remote_clusters_remote_clusters·js/Kibana_Pipeline___kibana_xpack_agent___apis_management_remote_clusters_Remote_Clusters_Add_should_not_allow_us_to_re_add_an_existing_remote_cluster)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook
[00:04:46]           └-: management
[00:04:46]             └-> "before all" hook
[00:05:00]             └-: remote clusters
[00:05:00]               └-> "before all" hook
[00:05:00]               └-: Remote Clusters
[00:05:00]                 └-> "before all" hook
[00:05:00]                 └-: Add
[00:05:00]                   └-> "before all" hook
[00:05:00]                   └-> should allow us to add a remote cluster
[00:05:00]                     └-> "before each" hook: global before each
[00:05:01]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-debian-tests-xl-1580921653932390078] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:01]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-debian-tests-xl-1580921653932390078] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:01]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-debian-tests-xl-1580921653932390078] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:01]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-debian-tests-xl-1580921653932390078] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:01]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-debian-tests-xl-1580921653932390078] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:01]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-debian-tests-xl-1580921653932390078] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:01]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-debian-tests-xl-1580921653932390078] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:01]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-debian-tests-xl-1580921653932390078] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:01]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-debian-tests-xl-1580921653932390078] updating [cluster.remote.test_cluster.skip_unavailable] from [false] to [true]
[00:05:01]                     └- ✓ pass  (82ms) "apis management remote clusters Remote Clusters Add should allow us to add a remote cluster"
[00:05:01]                   └-> should not allow us to re-add an existing remote cluster
[00:05:01]                     └-> "before each" hook: global before each
[00:05:01]                     └- ✖ fail: "apis management remote clusters Remote Clusters Add should not allow us to re-add an existing remote cluster"
[00:05:01]                     │

```

#### Stack Trace
```
Error: expected 409 "Conflict", got 400 "Bad Request"
    at Test._assertStatus (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:268:12)
    at Test._assertFunction (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:283:11)
    at Test.assert (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:173:18)
    at assert (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:131:12)
    at /dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:128:5
    at Test.Request.callback (/dev/shm/workspace/kibana/node_modules/superagent/lib/node/index.js:718:3)
    at parser (/dev/shm/workspace/kibana/node_modules/superagent/lib/node/index.js:906:18)
    at IncomingMessage.res.on (/dev/shm/workspace/kibana/node_modules/superagent/lib/node/parsers/json.js:19:7)
    at endReadableNT (_stream_readable.js:1143:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```
</details>

---
and 1 more failures, only showing the first 3.

### History
* :broken_heart: [Build #24390](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/) failed 03d8aa15abdbe5914f8b5604cdf27269b95977ef

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/","number":"24390","commit":"03d8aa15abdbe5914f8b5604cdf27269b95977ef"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/","number":"24644","commit":"0021d90befbbd1f2ebdaabb6040435d4c9329cba"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/)
* Commit: 6fd2c7f445c25548a60d8f26332940c01aa76323

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/testReport)

<details><summary>Kibana Pipeline / kibana-xpack-agent / X-Pack API Integration Tests.x-pack/test/api_integration/apis/management/remote_clusters/remote_clusters·js.apis management remote clusters Remote Clusters Add should not allow us to re-add an existing remote cluster</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/testReport//junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_management_remote_clusters_remote_clusters·js/Kibana_Pipeline___kibana_xpack_agent___apis_management_remote_clusters_Remote_Clusters_Add_should_not_allow_us_to_re_add_an_existing_remote_cluster)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook
[00:04:17]           └-: management
[00:04:17]             └-> "before all" hook
[00:04:31]             └-: remote clusters
[00:04:31]               └-> "before all" hook
[00:04:31]               └-: Remote Clusters
[00:04:31]                 └-> "before all" hook
[00:04:31]                 └-: Add
[00:04:31]                   └-> "before all" hook
[00:04:31]                   └-> should allow us to add a remote cluster
[00:04:31]                     └-> "before each" hook: global before each
[00:04:31]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:04:31]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:04:31]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:04:31]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:04:31]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:04:31]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:04:31]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:04:31]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:04:31]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.skip_unavailable] from [false] to [true]
[00:04:31]                     └- ✓ pass  (65ms) "apis management remote clusters Remote Clusters Add should allow us to add a remote cluster"
[00:04:31]                   └-> should not allow us to re-add an existing remote cluster
[00:04:31]                     └-> "before each" hook: global before each
[00:04:31]                     └- ✖ fail: "apis management remote clusters Remote Clusters Add should not allow us to re-add an existing remote cluster"
[00:04:31]                     │

```

#### Stack Trace
```
Error: expected 409 "Conflict", got 400 "Bad Request"
    at Test._assertStatus (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:268:12)
    at Test._assertFunction (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:283:11)
    at Test.assert (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:173:18)
    at assert (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:131:12)
    at /dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:128:5
    at Test.Request.callback (/dev/shm/workspace/kibana/node_modules/superagent/lib/node/index.js:718:3)
    at parser (/dev/shm/workspace/kibana/node_modules/superagent/lib/node/index.js:906:18)
    at IncomingMessage.res.on (/dev/shm/workspace/kibana/node_modules/superagent/lib/node/parsers/json.js:19:7)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```
</details>

---

<details><summary>Kibana Pipeline / kibana-xpack-agent / X-Pack API Integration Tests.x-pack/test/api_integration/apis/management/remote_clusters/remote_clusters·js.apis management remote clusters Remote Clusters Add should not allow us to re-add an existing remote cluster</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/testReport//junit/X-Pack%20API%20Integration%20Tests/x-pack_test_api_integration_apis_management_remote_clusters_remote_clusters·js/Kibana_Pipeline___kibana_xpack_agent___apis_management_remote_clusters_Remote_Clusters_Add_should_not_allow_us_to_re_add_an_existing_remote_cluster_2)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches

[00:00:00]       │
[00:00:00]         └-: apis
[00:00:00]           └-> "before all" hook
[00:04:55]           └-: management
[00:04:55]             └-> "before all" hook
[00:05:09]             └-: remote clusters
[00:05:09]               └-> "before all" hook
[00:05:09]               └-: Remote Clusters
[00:05:09]                 └-> "before all" hook
[00:05:09]                 └-: Add
[00:05:09]                   └-> "before all" hook
[00:05:09]                   └-> should allow us to add a remote cluster
[00:05:09]                     └-> "before each" hook: global before each
[00:05:09]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:09]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:09]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:09]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:09]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:09]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:09]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:09]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.seeds] from [[]] to [["localhost:6173"]]
[00:05:09]                     │ info [o.e.c.s.ClusterSettings] [kibana-ci-immutable-centos-tests-xl-1581024350419232108] updating [cluster.remote.test_cluster.skip_unavailable] from [false] to [true]
[00:05:09]                     └- ✓ pass  (71ms) "apis management remote clusters Remote Clusters Add should allow us to add a remote cluster"
[00:05:09]                   └-> should not allow us to re-add an existing remote cluster
[00:05:09]                     └-> "before each" hook: global before each
[00:05:09]                     └- ✖ fail: "apis management remote clusters Remote Clusters Add should not allow us to re-add an existing remote cluster"
[00:05:09]                     │

```

#### Stack Trace
```
Error: expected 409 "Conflict", got 400 "Bad Request"
    at Test._assertStatus (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:268:12)
    at Test._assertFunction (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:283:11)
    at Test.assert (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:173:18)
    at assert (/dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:131:12)
    at /dev/shm/workspace/kibana/node_modules/supertest/lib/test.js:128:5
    at Test.Request.callback (/dev/shm/workspace/kibana/node_modules/superagent/lib/node/index.js:718:3)
    at parser (/dev/shm/workspace/kibana/node_modules/superagent/lib/node/index.js:906:18)
    at IncomingMessage.res.on (/dev/shm/workspace/kibana/node_modules/superagent/lib/node/parsers/json.js:19:7)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```
</details>

---

### History
* :broken_heart: [Build #24644](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/) failed 0021d90befbbd1f2ebdaabb6040435d4c9329cba
* :broken_heart: [Build #24390](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/) failed 03d8aa15abdbe5914f8b5604cdf27269b95977ef

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/","number":"24390","commit":"03d8aa15abdbe5914f8b5604cdf27269b95977ef"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/","number":"24644","commit":"0021d90befbbd1f2ebdaabb6040435d4c9329cba"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/","number":"25085","commit":"6fd2c7f445c25548a60d8f26332940c01aa76323"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25155/)
* Commit: c9d0c5db05abfa3fdebe8b14129feb239a38e82f

### History
* :broken_heart: [Build #25085](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/) failed 6fd2c7f445c25548a60d8f26332940c01aa76323
* :broken_heart: [Build #24644](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/) failed 0021d90befbbd1f2ebdaabb6040435d4c9329cba
* :broken_heart: [Build #24390](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/) failed 03d8aa15abdbe5914f8b5604cdf27269b95977ef

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/","number":"24390","commit":"03d8aa15abdbe5914f8b5604cdf27269b95977ef"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/","number":"24644","commit":"0021d90befbbd1f2ebdaabb6040435d4c9329cba"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/","number":"25085","commit":"6fd2c7f445c25548a60d8f26332940c01aa76323"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25155/","number":"25155","commit":"c9d0c5db05abfa3fdebe8b14129feb239a38e82f"}]}
PIPELINE-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25238/)
* Commit: 14cbcb6cb1825fc89c6907a0d34b3231956158c6

### History
* :broken_heart: [Build #25155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25155/) failed c9d0c5db05abfa3fdebe8b14129feb239a38e82f
* :broken_heart: [Build #25085](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/) failed 6fd2c7f445c25548a60d8f26332940c01aa76323
* :broken_heart: [Build #24644](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/) failed 0021d90befbbd1f2ebdaabb6040435d4c9329cba
* :broken_heart: [Build #24390](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/) failed 03d8aa15abdbe5914f8b5604cdf27269b95977ef

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/","number":"24390","commit":"03d8aa15abdbe5914f8b5604cdf27269b95977ef"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/","number":"24644","commit":"0021d90befbbd1f2ebdaabb6040435d4c9329cba"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/","number":"25085","commit":"6fd2c7f445c25548a60d8f26332940c01aa76323"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25155/","number":"25155","commit":"c9d0c5db05abfa3fdebe8b14129feb239a38e82f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25238/","number":"25238","commit":"14cbcb6cb1825fc89c6907a0d34b3231956158c6"}]}
PIPELINE-->

jloleysens commented on x-pack/legacy/plugins/remote_clusters/index.ts:
```diff
@@ -7,8 +7,6 @@
 import { Legacy } from 'kibana';
```
I think if the goal of this contribution is to complete the NP migration we should strip this file down to the bare minimum required to keep functional parity with legacy. For instance:

- Migrating `config` and `injectDefaultVars` see https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#configure-plugin
- Similarly `configPrefix` can be removed if we define it in `kibana.json` I believe
- If we use the new platform management service for registering the plugin can we remove `managementSections`?
- Is `isEnabled` being used somewhere else?

alisonelizabeth replied:
👍 I was planning on addressing these items in a separate PR as part of the `public` migration. 

Also, `isEnabled` is being used by CCR so I think this needs to stay as is for now.

jloleysens commented on x-pack/plugins/remote_clusters/server/routes/api/add_route.ts:
```diff
@@ -0,0 +1,90 @@
+import { RouteDependencies } from '../../types';
+
+export const register = (deps: RouteDependencies): void => {
+  const addHandler: RequestHandler<any, any, any> = async (ctx, request, response) => {
```
I think it would be good to try and avoid use of `any` here when we know what the query, params and body will be.

Using `unknown` explicitly when we know we don't expect a value in one of those slots makes it clear what we expect for a route. Then passing in type information from:

```ts
body: schema.object({
          name: schema.string(),
          seeds: schema.arrayOf(schema.string()),
          skipUnavailable: schema.boolean(),
        }),
```

would be really nice!

alisonelizabeth replied:
Good point, fixed!

jloleysens commented on x-pack/plugins/remote_clusters/server/routes/api/delete_route.ts:
```diff
@@ -0,0 +1,129 @@
+          const cluster = get(updateClusterResponse, `persistent.cluster.remote.${name}`);
+
+          if (acknowledged && !cluster) {
+            return null;
```
A short comment here explaining why we are not returning a `KibanaResponse` for the happy path would be helpful 🙂

alisonelizabeth replied:
Added a comment. The response is being returned on [L104](https://github.com/elastic/kibana/pull/56781/files/14cbcb6cb1825fc89c6907a0d34b3231956158c6#diff-b0d61bb65a509d3d026dec9198a72ab0R104).

alisonelizabeth commented:
@jloleysens I believe I addressed all of your comments. Mind taking another look? I plan on creating a separate PR to complete the migration of the `public` code and anything else outstanding in `x-pack/legacy/plugins/remote_clusters/index.ts`

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25603/)
* Commit: 276389aa000ac85544efbbdd10b5540497a8628a

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25603/testReport)

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/plugins/spaces/server/lib/request_interceptors.onPostAuthInterceptor requests proxied to the legacy platform redirects to the space selector screen when accessing an app within a non-existent space</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25603/testReport//junit/X-Pack%20Jest%20Tests/x-pack_plugins_spaces_server_lib_request_interceptors/Kibana_Pipeline___x_pack_intake_agent___onPostAuthInterceptor_requests_proxied_to_the_legacy_platform_redirects_to_the_space_selector_screen_when_accessing_an_app_within_a_non_existent_space)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 8 times on tracked branches: https://github.com/elastic/kibana/issues/55953


```

#### Stack Trace
```
: Timeout - Async callback was not invoked within the 30000ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 30000ms timeout specified by jest.setTimeout.Error: 
    at new Spec (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
    at new Spec (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/setup_jest_globals.js:80:9)
    at specFactory (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/Env.js:575:24)
    at Env.it (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/Env.js:644:24)
    at Env.it (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:132:23)
    at it (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/jasmineLight.js:93:21)
    at Suite.describe (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.test.ts:233:5)
    at addSpecsToSuite (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/Env.js:496:51)
    at Env.describe (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/Env.js:466:11)
    at describe (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/jasmineLight.js:81:18)
    at Suite.Object.<anonymous>.describe (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.test.ts:232:3)
    at addSpecsToSuite (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/Env.js:496:51)
    at Env.describe (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/Env.js:466:11)
    at describe (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/jasmine/jasmineLight.js:81:18)
    at Object.<anonymous> (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/x-pack/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.test.ts:33:1)
    at Runtime._execModule (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runtime/build/index.js:867:68)
    at Runtime._loadModule (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runtime/build/index.js:577:12)
    at Runtime.requireModule (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runtime/build/index.js:433:10)
    at /var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:202:13
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:27:24)
    at _next (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:47:9)
    at /var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:52:7
    at new Promise (<anonymous>)
    at /var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:44:12
    at jasmine2 (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-jasmine2/build/index.js:60:19)
    at /var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runner/build/runTest.js:385:24
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runner/build/runTest.js:161:24)
    at _next (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/node_modules/jest-runner/build/runTest.js:181:9)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

---

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/plugins/spaces/server/lib/request_interceptors.onPostAuthInterceptor requests proxied to the legacy platform when accessing the kibana app it always allows the request to continue</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25603/testReport//junit/X-Pack%20Jest%20Tests/x-pack_plugins_spaces_server_lib_request_interceptors/Kibana_Pipeline___x_pack_intake_agent___onPostAuthInterceptor_requests_proxied_to_the_legacy_platform_when_accessing_the_kibana_app_it_always_allows_the_request_to_continue)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 8 times on tracked branches: https://github.com/elastic/kibana/issues/55954


```

#### Stack Trace
```
Error: Http server is not setup up yet
    at HttpServer.start (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/src/core/server/http/http_server.ts:142:13)
    at HttpService.start (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/src/core/server/http/http_service.ts:134:29)
```
</details>

---

### History
* :green_heart: [Build #25238](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25238/) succeeded 14cbcb6cb1825fc89c6907a0d34b3231956158c6
* :broken_heart: [Build #25155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25155/) failed c9d0c5db05abfa3fdebe8b14129feb239a38e82f
* :broken_heart: [Build #25085](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/) failed 6fd2c7f445c25548a60d8f26332940c01aa76323
* :broken_heart: [Build #24644](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/) failed 0021d90befbbd1f2ebdaabb6040435d4c9329cba
* :broken_heart: [Build #24390](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/) failed 03d8aa15abdbe5914f8b5604cdf27269b95977ef

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24390/","number":"24390","commit":"03d8aa15abdbe5914f8b5604cdf27269b95977ef"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/","number":"24644","commit":"0021d90befbbd1f2ebdaabb6040435d4c9329cba"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/","number":"25085","commit":"6fd2c7f445c25548a60d8f26332940c01aa76323"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25155/","number":"25155","commit":"c9d0c5db05abfa3fdebe8b14129feb239a38e82f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25238/","number":"25238","commit":"14cbcb6cb1825fc89c6907a0d34b3231956158c6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25603/","number":"25603","commit":"276389aa000ac85544efbbdd10b5540497a8628a"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25628/)
* Commit: 4c722fed5e151ef8fe02752430d1915cb4a3bbee

### History
* :broken_heart: [Build #25603](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25603/) failed 276389aa000ac85544efbbdd10b5540497a8628a
* :green_heart: [Build #25238](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25238/) succeeded 14cbcb6cb1825fc89c6907a0d34b3231956158c6
* :broken_heart: [Build #25155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25155/) failed c9d0c5db05abfa3fdebe8b14129feb239a38e82f
* :broken_heart: [Build #25085](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/) failed 6fd2c7f445c25548a60d8f26332940c01aa76323
* :broken_heart: [Build #24644](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/) failed 0021d90befbbd1f2ebdaabb6040435d4c9329cba

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/24644/","number":"24644","commit":"0021d90befbbd1f2ebdaabb6040435d4c9329cba"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/","number":"25085","commit":"6fd2c7f445c25548a60d8f26332940c01aa76323"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25155/","number":"25155","commit":"c9d0c5db05abfa3fdebe8b14129feb239a38e82f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25238/","number":"25238","commit":"14cbcb6cb1825fc89c6907a0d34b3231956158c6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25603/","number":"25603","commit":"276389aa000ac85544efbbdd10b5540497a8628a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25628/","number":"25628","commit":"4c722fed5e151ef8fe02752430d1915cb4a3bbee"}]}
PIPELINE-->

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25689/)
* Commit: a65869990df2a3c6cd8a22b7f9d8c627198f453e

### History
* :broken_heart: [Build #25628](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25628/) failed 4c722fed5e151ef8fe02752430d1915cb4a3bbee
* :broken_heart: [Build #25603](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25603/) failed 276389aa000ac85544efbbdd10b5540497a8628a
* :green_heart: [Build #25238](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25238/) succeeded 14cbcb6cb1825fc89c6907a0d34b3231956158c6
* :broken_heart: [Build #25155](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25155/) failed c9d0c5db05abfa3fdebe8b14129feb239a38e82f
* :broken_heart: [Build #25085](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/) failed 6fd2c7f445c25548a60d8f26332940c01aa76323

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25085/","number":"25085","commit":"6fd2c7f445c25548a60d8f26332940c01aa76323"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25155/","number":"25155","commit":"c9d0c5db05abfa3fdebe8b14129feb239a38e82f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25238/","number":"25238","commit":"14cbcb6cb1825fc89c6907a0d34b3231956158c6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25603/","number":"25603","commit":"276389aa000ac85544efbbdd10b5540497a8628a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25628/","number":"25628","commit":"4c722fed5e151ef8fe02752430d1915cb4a3bbee"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/25689/","number":"25689","commit":"a65869990df2a3c6cd8a22b7f9d8c627198f453e"}]}
PIPELINE-->
